### PR TITLE
chore(ui): convert LoadingIndicator to Typescript

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

- Migrated the `LoadingIndicator` component to `TypeScript`, enhancing code quality, maintainability, and scalability.
- Utilised static typing to minimize runtime errors and improve the development experience.

# Changes Made

**TypeScript Migration**:

- Converted `LoadingIndicator` component files to `TypeScript`.
- Added comprehensive type annotations.

**Fixes/Improvements**:

- Allowed type `number` and `string` for `size`.
- Fixed edge cases for `size`, defaulting to size `96` when an _invalid_ number value (e.g. _empty_ string), `0` or a _negative_ value is passed.
- (Not related to ticket) Exported deprecated `StyleProvider` from ui-components/src `index` file.

**Testing**:

- Tested fixed edge cases.
- Verified component behaviour using `Storybook`.
- Updated tests to use `vitest`.

**Cleanup:**

- Removed deprecated code. It wasn't needed for any `.js` components.

**Documentation:**

- Updated `Storybook` documentation.

Screenshot: Deprecated `.js` code on the left, migrated `.ts` code on the right (**see notes**)

Screenshot 2024-10-25 at 09 14 09

**Release:**

- Generated a `changeset` file, mentioning the migration and bug fix.

# Related Issues

- https://github.com/cloudoperators/juno/issues/238

# Testing Instructions

1. Run `Storybook`
2. Run `LoadingIndicator` tests

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/components/LoadingIndicator/LoadingIndicator.component.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+import React from "react"
+import Loading from "./loading-indicator.svg"
+
+export interface LoadingIndicatorProps {
+  /**
+   * The size of the LoadingIndicator in pixels.
+   */
+  size?: string
```
Comment (guoda-puidokaite) [inline packages/ui-components/src/components/LoadingIndicator/LoadingIndicator.component.tsx:13]: Few questions:
- Should we allow `size` as a `number` as well and account for that in the code? Is there a reason it's only a `string`?
- We expect the `size` to be passed as a `string`, without `px` included. What if the user does include `px` (or something else entirely) as part of the input? Is that fine? Or should we filter it out? Allowing only `number` type would really help here (if we indeed only want that)... but I understand it's not so easy to change it. Would just love to understand the constraints so we can cover edge cases here. 💪 Maybe I misunderstood something though. 

CC: @edda

Comment (edda) [reply packages/ui-components/src/components/LoadingIndicator/LoadingIndicator.component.tsx:13]: Those are good questions. I can't tell you why we made the choice of going with string here. I agree that going with number type instead would take care of a bunch of edge cases because people can only put numbers in, so there'll never be the edge case of people accidentally passing the unit together with the number.

The only thing I can think of is that the svg component requires the width/height to be a string but that can easily be solved by turning the passed number into a string. So I agree with you, let's make it number instead of string.

Comment (guoda-puidokaite) [reply packages/ui-components/src/components/LoadingIndicator/LoadingIndicator.component.tsx]: Perfect, Thank You so much. Noted it here: https://github.com/cloudoperators/juno/issues/234#issuecomment-2435175493 and I'll create a ticket for future improvements, as discussed. 🙏

**Review by barsukov — APPROVED** [review]
LGTM thank you for little fix in index

**Review by barsukov — CHANGES_REQUESTED** [review]
```
ERROR in ./src/components/LoadingIndicator/LoadingIndicator.stories.tsx 5:0-46
Module not found: Error: Can't resolve './index.js' in '/Users/wowa/work/juno/packages/ui-components/src/components/LoadingIndicator'
```

**Comment by guoda-puidokaite** [comment]
> ```
> ERROR in ./src/components/LoadingIndicator/LoadingIndicator.stories.tsx 5:0-46
> Module not found: Error: Can't resolve './index.js' in '/Users/wowa/work/juno/packages/ui-components/src/components/LoadingIndicator'
> ```

Ooops, somehow missed to commit that file... Pushed now.

**Review by barsukov — APPROVED** [review]

**Review by edda — APPROVED** [review]
Love the edge cases fixing 🥇 
I made two small changes to comments (took out CCloud reference, fixed color class name reference)

**Review by barsukov — APPROVED** [review]
